gdnative-derive/src/methods.rs

```diff
@@ -196,122 +221,138 @@ fn impl_gdnative_expose(ast: ItemImpl) -> (ItemImpl, ClassMethodExport) {
-                                        let msg = format!(
-                                            "unexpected attribute argument: {}",
-                                            meta.into_token_stream()
-                                        );
+                        if is_export {
+                            let mut _export_args =
+                                export_args.get_or_insert_with(ExportArgs::default);
+                            _export_args.is_old_syntax = is_old_syntax;
+                            use syn::{punctuated::Punctuated, Lit, Meta, NestedMeta};
+                            let nested_meta_iter = match attr.parse_meta() {
+                                Err(err) => {
+                                    errors.push(err);
+                                    return false;
+                                }
+                                Ok(Meta::NameValue(name_value)) => {
+                                    let span = name_value.span();
+                                    let msg = "NameValue syntax is not valid";
+                                    errors.push(syn::Error::new(span, msg));
+                                    return false;
+                                }
+                                Ok(Meta::Path(_)) => {
+                                    Punctuated::<NestedMeta, syn::token::Comma>::new().into_iter()
+                                }
+                                Ok(Meta::List(list)) => list.nested.into_iter(),
+                            };
+                            for nested_meta in nested_meta_iter {
+                                let (path, lit) = match &nested_meta {
+                                    NestedMeta::Lit(param) => {
+                                        let span = param.span();
+                                        let msg = "Literal item is not valid";
                                         errors.push(syn::Error::new(span, msg));
-                                        return false;
+                                        continue;
                                     }
+                                    NestedMeta::Meta(param) => match param {
+                                        Meta::List(list) => {
+                                            let span = list.span();
+                                            let msg = "List item is not valid";
+                                            errors.push(syn::Error::new(span, msg));
+                                            continue;
+                                        }
+                                        Meta::Path(path) => (path, None),
+                                        Meta::NameValue(name_value) => {
+                                            (&name_value.path, Some(&name_value.lit))
+                                        }
+                                    },
                                 };
```

Comment:
What are all these changes doing (see top-level comment)?

Reply:
Syntax parsing of macro parameters. ([detailed syntax](https://doc.rust-lang.org/reference/attributes.html#meta-item-attribute-syntax)) 

Codes like `#[macro(path, name = "value")]` are accepted. Codes like `#[path]`, `#[name = "value"]` or `#[macro("lit")]` are not accepted.

Reply:
Thank you!

> Codes like `#[macro(path, name = "value")]` are accepted. Codes like `#[path]`, `#[name = "value"]` or `#[macro("lit")]` are not accepted.

Could you add this as a comment?

Reply:
Added comments.